shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -736,8 +737,13 @@ public String getDartEntrypointFunctionName() {
    */
   @NonNull
   public String getInitialRoute() {
-    if (getIntent().hasExtra(EXTRA_INITIAL_ROUTE)) {
-      return getIntent().getStringExtra(EXTRA_INITIAL_ROUTE);
+    Intent intent = getIntent();
+    Uri data = intent.getData();
```

Comment:
The developer still need to manually modify the AndroidManifest to add an intent filter. I wonder if we can automatically generate the intent filter for them so it will be much simpler for flutter beginner to work with deeplink.

Reply:
You can change the default create template's AndroidManifest if I understood